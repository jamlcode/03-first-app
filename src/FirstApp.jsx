// variables que no tengan dependencias dentro del componente dejarlas fuera
// const newMessage = {
//   title: 'hola',
//   body: 'mundo',
//   uuid: '123'
// }
import PropTypes from 'prop-types';  

export default function FirstApp({
  title,
  state = {
    value: 'inactive',
    label: 'Inactivo'
  },
  subtitle,
}) {
  return (
    <main className='container'>
      <h1>{title}</h1>
      <h2 data-testid='test-subtitle' > {subtitle} </h2>
      <span style={{ backgroundColor: state.value === 'active' ? 'green' : 'red' }} >{state.label}</span>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem id debitis ad consequuntur, nulla voluptates deserunt voluptatibus consequatur rerum incidunt voluptatum soluta accusantium. Itaque, ipsum nostrum enim voluptate porro doloribus.
      </p>
    </main>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  state: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired
  }),
  subtitle: PropTypes.string,
}

/**
 * Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead
 */
// FirstApp.defaultProps = {
//   title: 'No hay tiÃÅtulo',
//   state: {
//     value: 'inactive',
//     label: 'Inactivo'
//   },
//   subtitle: 'Goku'
// }