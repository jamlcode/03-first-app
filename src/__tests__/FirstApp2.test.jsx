import { describe, it, expect, afterEach } from "vitest";
import { render, cleanup, screen } from "@testing-library/react";
import FirstApp from "../FirstApp";

const title = 'FirstApp';
const subTitle = 'Goku';
describe('Test in <FirstApp/>', () => {
    // each test render the component so we need to clean it before each test
    afterEach(cleanup)
        

    it('should match with snapshot', () => {
        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    })
    it('should show the title', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy();
    })
    it('should show the title inside h1 tag', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    })
    it('should show the subtitle inside h2 tag', () => {
        render(<FirstApp title={title} subtitle={subTitle} />)
        expect(screen.getByTestId('test-subtitle')).toBeTruthy();
        expect(screen.getByTestId('test-subtitle').innerHTML).toContain(subTitle)
    })
})