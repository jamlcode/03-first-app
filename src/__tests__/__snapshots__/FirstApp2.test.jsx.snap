// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test in <FirstApp/> > should match with snaps hot 1`] = `
<div>
  <main
    class="container"
  >
    <h1>
      FirstApp
    </h1>
    <h2
      data-testid="test-subtitle"
    >
       
       
    </h2>
    <span
      style="background-color: red;"
    >
      Inactivo
    </span>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem id debitis ad consequuntur, nulla voluptates deserunt voluptatibus consequatur rerum incidunt voluptatum soluta accusantium. Itaque, ipsum nostrum enim voluptate porro doloribus.
    </p>
  </main>
</div>
`;

exports[`Test in <FirstApp/> > should match with snapshot 1`] = `
<div>
  <main
    class="container"
  >
    <h1>
      FirstApp
    </h1>
    <h2
      data-testid="test-subtitle"
    >
       
       
    </h2>
    <span
      style="background-color: red;"
    >
      Inactivo
    </span>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem id debitis ad consequuntur, nulla voluptates deserunt voluptatibus consequatur rerum incidunt voluptatum soluta accusantium. Itaque, ipsum nostrum enim voluptate porro doloribus.
    </p>
  </main>
</div>
`;
